api/src/services/telemetry-services/telemetry-device-service.ts

```diff
@@ -59,6 +72,61 @@ export class TelemetryDeviceService extends DBService {
     return this.telemetryDeviceRepository.getDevicesByIds(surveyId, deviceIds);
   }
 
+  /**
+   * Get all devices for a survey, based on pagination options.
+   *
+   * @param {number} surveyId
+   * @param {ApiPaginationOptions} [pagination]
+   * @return {*}  {Promise<DeviceRecord[]>}
+   * @memberof TelemetryDeviceService
+   */
+  async getDevicesForSurvey(surveyId: number, pagination?: ApiPaginationOptions): Promise<DeviceRecord[]> {
+    return this.telemetryDeviceRepository.getDevicesForSurvey(surveyId, pagination);
+  }
+
+  /**
+   * Find devices.
+   *
+   * @param {boolean} isUserAdmin Whether the user is an admin.
+   * @param {(number | null)} systemUserId The user's ID.
+   * @param {DeviceAdvancedFilters} filterFields The filter fields to apply.
+   * @param {ApiPaginationOptions} [pagination] The pagination options.
+   * @return {*}  {Promise<DeviceRecord[]>}
+   * @memberof TelemetryDeviceService
+   *
+   */
+  async findDevices(
+    isUserAdmin: boolean,
+    systemUserId: number | null,
+    filterFields: DeviceAdvancedFilters,
+    pagination?: ApiPaginationOptions
+  ): Promise<DeviceRecord[]> {
+    return this.telemetryDeviceRepository.findDevices(isUserAdmin, systemUserId, filterFields, pagination);
+  }
+
+  /**
+   * Get the total count of all devices for a survey.
+   *
+   * @param {number} surveyId
+   * @return {*}  {Promise<number>}
+   * @memberof TelemetryDeviceService
+   */
+  async getDevicesCount(surveyId: number): Promise<number> {
+    return this.telemetryDeviceRepository.getDevicesCount(surveyId);
```

Comment:
question: Couldn't we just use the length of getDevicesForSurvey()?

Reply:
tbh, all of our "get all X by survey id" should really include pagination support. I think most do now? Though I don't think I did that for the get all deployments endpoint, so I might add that